src/Parser/RichParser.php

```diff
@@ -66,6 +66,9 @@ public function parseString(string $sourceCode): array
 		$traitCollectingVisitor = new TraitCollectingVisitor();
 		$nodeTraverser->addVisitor($traitCollectingVisitor);
 
+		$transformElseIfToFlatElseIfVisitor = new TransformElseIfToFlatElseIfVisitor();
+		$nodeTraverser->addVisitor($transformElseIfToFlatElseIfVisitor);
```

Comment:
I'm a bit surprised this works. I'm not sure about the order and effects the visitors are called in. I'd expect the other visitors not being able to see effects from this one.

I'd be a little bit calmer if the AST was transformed with `TransformElseIfToFlatElseIfVisitor` by a separate NodeTraverser and only after that it was passed to the current `$nodeTraverser`.